src/coreclr/jit/gentree.cpp

```diff
@@ -20059,6 +20059,24 @@ GenTree* Compiler::gtNewSimdCmpOpNode(genTreeOps  op,
 
             if (intrinsic == NI_Illegal)
             {
+                // If we don't have an intrinsic set for this, try "Min(op1, op2) == op1"
+                if (((simdSize == 32) && compOpportunisticallyDependsOn(InstructionSet_AVX2)) ||
```

Comment:
Might be worth a comment that we don't specialize `byte` for SSE2 only since it complicates our support matrix for a much smaller set of hardware.

It would benefit R2R, but hot methods will get re-optimized anyways so the 1-2 cycle difference isn't worth it.